### PR TITLE
Add a custom_content block for user templates

### PR DESCRIPTION
**Pull request by b-butler**
## Description

Adds a jinja block specifically for user additions.

## Motivation and Context

This is part of the discussion @bdice and a few others and I had about how to make user customizations to templates easier.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/templates/custom_script_template.sh [hunk 1]
```diff
@@ -9,6 +9,13 @@ custom commands (e.g. `module load ...`).
 {% block header %}
     {{- super () -}}
 {% endblock header %}
+{% block custom_content %}
+{#
+    This block is not used by any other template and can be safely modified
+    without the need to call super(). We recommend most additions to the
+    templates go here if they are not direct changes to an existing template.
+#}
```
Comment (bdice) [inline flow/templates/custom_script_template.sh:17]:
```suggestion
{#
    This block is not used by any other template and can be safely modified
    without the need to call super(). We recommend most additions to the
    templates go here if they are not direct changes to an existing template.
    
    For example, commands like `module load ...` or printing diagnostic
    information from the scheduler can be done in this block.
#}
```

Comment (b-butler) [reply flow/templates/custom_script_template.sh:17]: I like it!

**Review by bdice — APPROVED** [review]
LGTM!

**Review by lyrivera — APPROVED** [review]
LGTM too.

**Pull request merged**